SparkPost/2web2ui
# SA-2069 - (UI) Adds toggle for access to deliverability data in subaccounts details page.

[SA-2069](https://sparkpost.atlassian.net/browse/SA-2069)

### What Changed

- Adds supporting integration tests for the subaccounts create and edit pages (but only the details tab)
- Adds `ToggleBlock` Redux Form wrapper (😞 I wanted to refactor this using `react-hook-form` but have a bit too much going on - hate to add to our Redux Form code at this point but it is what it is)
- Incorporates toggle switch in to the `/account/subaccounts/<id>` page
- **Edit:** Updates the `GrantsCheckboxes` component:
  - Removes the `show` prop
  - Removes any CSS modules in favor of CSS grid to fix some layout bugs existing in the app currently
  - Removes OG/Hibana theme-related logic
  - Updates snapshot 😞 

### How To Test

- Go to the subaccount details page and then click on a subaccount to view the details page
- By default, the switch should match the server state for each subaccount in the "off" position
- When turned on, the switch should remain turned on when the page is refreshed
- Verify the correct data is being passed to the API. See the Jira ticket for relevant API docs.
- Ensure Cypress tests adequately capture behavior of the create and edit subaccount details pages
- Ensure Cypress tests do not have a high risk of flakiness
- **Edit:** Ensure the new UI does _not_ render unless the relevant product and account UI option are on the account

### To Do

- [ ] Does this switch need to be behind a product flag? Checking with product
